### PR TITLE
Enable HTTP Basic Auth to be set for the API.

### PR DESCRIPTION
**Pull request by smn**
Currently it's open to the world and that's a _really_ bad idea.
This allows one to set AUTH_USERNAME and AUTH_PASSWORD which'll 
have nginx set the `auth_basic` directive with the realm set to 
"Junebug" and add the username + crypted password to an htpasswd
file.

It'll also print a warning banner if people are running this docker
container without specifying the username and password environment 
variables.

The authentication of the various channels is the responsibility of the channel and channel configs.

### CODE REVIEW

**Comment by smn** [comment]
@JayH5 & @rudigiesler ready for review

**Comment by miltontony** [comment]
This looks good to me, but I'll leave it to other people with more context to signoff ~~:+1:~~

File: nginx-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+*                                                           *
+*************************************************************
+EOF
+fi
+
+cat > $NGINX_FILE <<EOF
+location /jb/ {
+  auth_basic ${AUTH_BASIC};
+  auth_basic_userfile ${HTPASSWD_FILE};
+  proxy_pass http://127.0.0.1:8080/;
```
Comment (jerith) [inline nginx-entrypoint.sh]: Should this use `$JUNEBUG_PORT` instead of `8080`?

File: nginx-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+EOF
+
+nginx -g "daemon off;"
```
Comment (jerith) [inline nginx-entrypoint.sh]: Do we want to `exec` this?

**Comment by smn** [comment]
@jerith thanks, ready for re-review.

**Review by jerith — APPROVED** [review]
Given that `$JUNEBUG_INTERFACE` and `$JUNEBUG_PORT` are only used behind niginx, do we even want them to be configurable?

:+1: either way.

**Comment by smn** [comment]
Good point, I don't know. Considering it's been exposed as an environment variable it only makes sense to respect whatever it's used in nginx. This would prevent breakages if people have changed it.

Thanks, merging.

**Pull request merged**